common/bolt12.c

```diff
@@ -106,12 +107,10 @@ static char *check_signature(const tal_t *ctx,
 	return NULL;
 }
 
-static const u8 *string_to_data(const tal_t *ctx,
-				const char *str,
-				size_t str_len,
-				const char *hrp_expected,
-				size_t *dlen,
-				char **fail)
+STATIC_UNLESS_FUZZING const u8 *string_to_data(const tal_t *ctx,
+					       const char *str, size_t str_len,
+					       const char *hrp_expected,
+					       size_t *dlen, char **fail)
```

Comment:
I really dislike this level of intrusiveness; we go to great lengths to avoid this in unit tests, we should do the same for fuzz tests.

In particular, fuzzing should not be afraid to include raw C files to get this kind of access.  It may mean you need to use generated mocks, like unit tests do, though?

Reply:
Done, PTAL.